msvc/scripts/genproj.cs

```diff
@@ -887,6 +945,26 @@ public VsCsproj Generate (string library_output, Dictionary<string,MsbuildGenera
 
 		bool basic_or_build = (library.Contains ("-basic") || library.Contains ("-build"));
 
+		// If an EXE is built with nostdlib, it won't work unless run with mono.exe. This stops our build steps
+		//  from working in visual studio (because we already replace @MONO@ with '' on Windows.)
+
+		if (Target != Target.Library)
+			StdLib = true;
+
+		// We have our target framework set to 4.5 in many places because broken scripts check for files with 4.5
+		//  in the path, even though we compile code that uses 4.6 features. So we need to manually fix that here.
+
+		if (fx_version == "4.5")
+			fx_version = "4.6.2";
+
+		// The VS2017 signing system fails to sign using this key for some reason, so for now,
+		//  just disable code signing for the nunit assemblies. It's not important.
+		// I'd rather fix this by updating the makefiles but it seems to be impossible to disable
+		//  code signing in our make system...
+
+		if ((StrongNameKeyFile != null) && StrongNameKeyFile.Contains("nunit.snk"))
```

Comment:
Is "File" a full path or a leaf? Either way feels a little bad to special case the string but surely ok for now, for a long time.

Reply:
Simpler `StrongNameKeyFile?.Contains("nunit.snk")` syntax